fead: add Aman_Asrani_gsoc24_final_report (#225)
CircuitVerse/Blog: content/posts/Aman_Asrani_gsoc24_final_report.md

### Project goals
---
So far, we have covered and implemented the following features:

- SAML Based SSO
- Email Verification During Registration for New Users 
- Send Email Verification button
- Migration to View Components
- Logs Maintenance
- Updation of Weekly Contest feature
- UI Improvements
- CircuitVerse Language Support

---

### SAML based SSO

Single sign-on (SSO) is an identification method that enables users to log in to multiple applications and websites with one set of credentials. SSO streamlines the authentication process for users.SAML (Security Assertion Markup Language) is a markup language designed for exchanging authentication information between the user, the identity provider, and the service provider.

> We initially encountered a number of problems, some of which were due to gem devise-saml-authenticable and gem ruby-saml versioning compatibility. Finally, we were able to redirect to the Identity Provider(IDP) login page and obtain the SAML answer after addressing these issues. However, we then ran into a problem while trying to sign the user, which took too much time. During a meeting with Tanmoy Bhaiya, we discovered that the create method in the saml_sessions_controller was being overridden by the create method from the devise_saml_authenticable gem. Finally, SAML-based SSO worked smoothly.

Reviewer: ```suggestion
We initially encountered a number of problems, some of which were due to gem devise-saml-authenticable and gem ruby-saml versioning compatibility. Finally, we were able to redirect to the Identity Provider(IDP) login page and obtain the SAML answer after addressing these issues. However, we then ran into a problem while trying to sign the user, which took too much time. During a meeting with Tanmoy, we discovered that the create method in the saml_sessions_controller was being overridden by the create method from the devise_saml_authenticable gem. Finally, SAML-based SSO worked smoothly.
```

It's a formal blog!

- Moved the footer links rendering into `FooterLinksComponent`
- Moved the social links rendering into `SocialLinksComponent`
- Integrated both components into the main `FooterComponent`
- Moved the footer rendering logic from the partial into `FooterComponent`

![Migrate footer partial into a view component](/images/Aman_Asrani/footerViewComponent.png)
![Migrate footer partial into a view component](/images/Aman_Asrani/final_report_gsoc2024/footer_Preview.png)

##### Pull Requests
- PR : [Migrate footer partial into a view component](https://github.com/CircuitVerse/CircuitVerse/pull/5011)

#### Migrate-SocialCard-to-view-component:
The component includes logic for image preview, title, and URL generation. Also added tests and previews with notes and parameters to demonstrate usage.
![Migrate-SocialCard-to-view-component](/images/Aman_Asrani/SocialCardViewComponent.png)
![Migrate-SocialCard-to-view-component](/images/Aman_Asrani/SocialCardViewComponentReuse.png)

##### Pull Requests
- PR : [Migrate-SocialCard-to-view-component](https://github.com/CircuitVerse/CircuitVerse/pull/5012)

### Logs Maintenance

> Initially, We had considered setting up a Git Hook. However, this approach would likely requirestoring the logs in our database, which is an expensive task.

Reviewer: ```suggestion
Initially, We had considered setting up a Git Hook. However, this approach requires storing the logs in our database, which is expensive.
```
